### DISCUSSION TITLE
How to get better fit of the data ?

### DISCUSSION THREAD

**Discussion by satyasaran** [MilesCranmer/PySR#602] [Q&A]
```python
#want to estimate    U_pred=f(x,t)

import sympy as sp
import numpy as np
import pandas as pd
from sklearn.preprocessing import MinMaxScaler
from pysr import PySRRegressor
from sklearn.model_selection import train_test_split

df1 = pd.read_csv("solitary.csv")

# Extract features and target variable
X_raw = df1[['x', 't']].values
y = df1['U_pred'].values # Normalize features using Min-Max scaling
scaler = MinMaxScaler()
X_scaled = scaler.fit_transform(X_raw)

# Fit the PySR model
model = PySRRegressor(
    niterations=50,
    batching=True,
    batch_size=264,
    elementwise_loss="L2EpsilonInsLoss(0.000001)",
    fast_cycle=True,
    binary_operators=["+", "*"],
    unary_operators=["cos", "exp", "sin", "quart(x) = x^4", "inv(x)=x^-1"],
    extra_sympy_mappings={"quart": lambda x: x**4, "inv": lambda x: 1/x},
    populations=40,
   
)
model.fit(X_scaled, y)

 # Predict using the model on the normalized features
y_pred = model.predict(X_scaled)

# Calculate R^2 score
r2 = r2_score(y, y_pred)
print("R^2 score:", r2)
#Output: R^2 score: 0.92. 
```
However, my goal is to achieve at least 0.98. Is this feasible? I noticed that the final model, after simplification, does not include the variable corresponding to 't'. Despite trying various loss functions, including the default and others from the documentation, the results remained largely unchanged. Any suggestions? I'm seeking a model in the form of 'u_pred = f(x,t)', but it consistently provides the best model as a function of 'x' alone

**Comment by MilesCranmer** [comment]
Check out https://astroautomata.com/PySR/tuning/
